app/ts/library/transactions.tsx

```diff
@@ -1,15 +1,23 @@
+import { TransactionResponse } from 'ethers';
 import { Contract, BrowserProvider } from 'ethers'
 import { ERC1155ABI, ERC721ABI } from './abi.js'
+import { humanReadableEthersError } from './humanEthersErrors.js';
 import { ERC1155, ERC721 } from './identifyTokens.js'
 
-export async function transferERC721(nft: ERC721, recipient: string, provider: BrowserProvider) {
+export async function transferERC721(nft: ERC721, recipient: string, provider: BrowserProvider): Promise<TransactionResponse> {
 	const contract = new Contract(nft.address, ERC721ABI, await provider.getSigner())
-	const tx = await contract.transferFrom(nft.owner, recipient, nft.id);
-	await tx.wait()
+	try {
+		return await contract.transferFrom(nft.owner, recipient, nft.id);
+	} catch (ethersError) {
+		throw humanReadableEthersError(ethersError)
+	}
 }
 
-export async function transferERC1155(nft: ERC1155, from: string, recipient: string, amount: bigint, provider: BrowserProvider) {
+export async function transferERC1155(nft: ERC1155, from: string, recipient: string, amount: bigint, provider: BrowserProvider): Promise<TransactionResponse> {
 	const contract = new Contract(nft.address, ERC1155ABI, await provider.getSigner())
-	const tx = await contract.safeTransferFrom(from, recipient, nft.id, amount, "0x");
-	await tx.wait()
+	try {
+		return await contract.safeTransferFrom(from, recipient, nft.id, amount, "0x");
+	} catch (ethersError) {
+		throw humanReadableEthersError(ethersError)
```

Comment:
here also its assumed its ethers error but it could be any other error too

Reply:
Moved to validating after asyncState has rejected.